lib/qesapdeployment.pm

```diff
@@ -438,6 +439,19 @@ sub qesap_ansible_log_find_timeout
     return $timeout_match ? 1 : 0;
 }
 
+=head3 qesap_ansible_log_find_missing_sudo_password
+
+    Return the '"msg": "Missing sudo password"' error found in the Ansible log or not
+=cut
+
+sub qesap_ansible_log_find_missing_sudo_password
+{
+    my ($file) = @_;
+    my $search_string = 'Missing sudo password';
+    my $timeout_match = script_output("grep \"$search_string\" $file || exit 0");
```

Comment:
You can also do:
```
my $missing_sudo_pwd_match = script_run("grep -q 'Missing sudo password' $file");
return !$missing_sudo_pwd_match;
```
`grep -q` would return 0 if the pattern is found, so you would just need to turn it into a Perl boolean by adding a `!`

I tend not to trust too much `script_output` for these cases, as you may get some garbled text in the result even if the `grep` did not match anything.

Reply:
Your comments are really good.
But I only did `s/timeout_match/missing_sudo_pwd_match/g` as if the code changed too much I have to do the VRs again and it is too hard to reproduce this error.